src/main/java/AddTodoCommand.java

```diff
@@ -0,0 +1,21 @@
+import java.io.IOException;
+
+public class AddTodoCommand extends Command {
+    private String taskDescription;
+
+    public AddTodoCommand(String description) {
+        this.taskDescription = description;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        Task t = tasks.addTodo(this.taskDescription);
+        try {
+            storage.save(tasks);
+        } catch (IOException e) {
+            // todo refactor throwing IOException
+            throw new DukeException("Error saving tasks");
```

Comment:
Don't we need to catch the thrown DukeException? :O

Reply:
I think the DukeException thrown here is passed onto and catched in line 56, run() function, of src/main/java/Duke.java. So it wouldn't be a problem. :)